pkg/kritis/cryptolib/verifier.go

```diff
@@ -49,13 +52,47 @@ type PublicKey struct {
 // NewPublicKey creates a new PublicKey. `keyType` contains the type of the
 // public key, one of Pgp, Pkix or Jwt. `keyData` contains the raw key
 // material. `keyID` contains a unique identifier for the public key. For PGP,
-// this should be the OpenPGP RFC4880 V4 fingerprint of the key.
-func NewPublicKey(keyType KeyType, keyData []byte, keyID string) PublicKey {
-	return PublicKey{
+// this should be the OpenPGP RFC4880 V4 fingerprint of the key. For PKIX and
+// JWT, the ID should contain valid URI characters.
+func NewPublicKey(keyType KeyType, keyData []byte, keyID string) (*PublicKey, error) {
+	switch keyType {
+	case Pgp:
+		err := validatePgpKeyID(keyData, keyID)
+		if err != nil {
+			return nil, err
+		}
+	case Pkix, Jwt:
+		// Valid URI characters (see http://tools.ietf.org/html/rfc3986#section-2)
+		reURI := regexp.MustCompile(`^[a-zA-Z0-9-._~:\/?#\[\]@!$&'\(\)*+,;=%]+$`)
```

Comment:
Is there any way to use url.Parse() for validation instead of this hard to understand regexp?

Reply:
Patrick advised me not to parse the key ID as a URI, but instead to check that it contains valid URI characters. He said that strong enforcement of URI-compatibility is deprecated in Binauthz.

See comment: https://docs.google.com/document/d/1u6V2pPrIrA4xFR3J2pcqjZajkuT5dZ8_Cck2co1prfg/edit?disco=AAAAGbNnOaM